followthemoney/types/number.py

```diff
@@ -22,9 +32,55 @@ class NumberType(PropertyType):
     def node_id(self, value: str) -> None:
         return None
 
+    def parse(
+        self, value: str, decimal: str = DECIMAL, separator: str = SEPARATOR
+    ) -> Tuple[Optional[str], Optional[str]]:
+        """Parse a number into a numeric value and a unit. The numeric value is
+        aligned with the decimal and separator settings. The unit is stripped of
+        whitespace and returned as a string. If no unit is found, None is
+        returned. If no number is found, None is returned for both values."""
+        value = value.replace(separator, "")
+        if decimal != self.DECIMAL:
+            value = value.replace(decimal, self.DECIMAL)
+        match = self.NUM_UNIT_RE.match(value)
+        if not match:
+            return None, None
+        number, unit = match.groups()
+        if unit is not None:
+            unit = unit.strip()
+            if len(unit) == 0:
+                unit = None
+        # TODO: We could have a lookup table for common units, e.g. kg, m, etc. to
+        # convert them to a standard form.
```

Comment:
As mentioned before, I think not checking whether a unit is actually valid for a given property is good enough, but I’m also not opposed to that idea, so if you think it would be useful/necessary, that’s absolutely fine from my perspective!